moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1990,21 +1990,22 @@ double RobotState::computeCartesianPath(const JointModelGroup* group, std::vecto
 }
 
 double RobotState::testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
-                                      const JumpThreshold& jump_threshold)
+                                      const JumpThreshold& jt)
 {
-  double percentage = 1.0;
-  if (jump_threshold.jump_threshold_factor > 0.0)
-    percentage *= testJointSpaceJump(group, traj, jump_threshold.jump_threshold_factor);
+  if (jt.jump_threshold_factor > 0.0)
+    return testJointSpaceJump(group, traj, jt.jump_threshold_factor);
 
-  if (jump_threshold.prismatic_jump_threshold > 0.0 || jump_threshold.revolute_jump_threshold > 0.0)
-    percentage *= testJointSpaceJump(group, traj, jump_threshold.revolute_jump_threshold,
-                                     jump_threshold.prismatic_jump_threshold);
+  else if (jt.prismatic_jump_threshold > 0.0 && jt.revolute_jump_threshold > 0.0)
+    return testJointSpaceJump(group, traj, jt.revolute_jump_threshold, jt.prismatic_jump_threshold);
 
-  return percentage;
+  else if (jt.prismatic_jump_threshold > 0.0 || jt.revolute_jump_threshold > 0.0)
```

Comment:
For me it doesn't make sense to test only revolute or only prismatic joints. Thus, both parameters need to be positive.

Reply:
I'm not sure if I agree with thuis. If I have a robot with only revolute joints, I don't want to have to pass in a made up prismatic_jump_threshold and vice versa

Reply:
I think one should provide reasonable thresholds for both cases. If you are very sure that you have no joints of either type, you can still pass any value > 0.

Reply:
I still don't like the idea of asking the user to make up prismatic joint cutoff values for their fully revolute robot. I just pushed a commit that deals with this issue in a way that I think is clean.

I also think that `else if` should be `if` to cover the use case when someone wants to be super safe and run both checks

Reply:
> I still don't like the idea of asking the user to make up prismatic joint cutoff values for their fully revolute robot. 

I guess we keep to have different opinions on that. What about the following: The thresholds are kept zero by default. If there is a joint of "wrong" type in the trajectory (although not expected), the zero threshold will reject the trajectory if this joint moves at all.
Then the absolute testing can be triggered `if (jt.prismatic > 0 || jt.revolute > 0)`

> I just pushed a commit that deals with this issue in a way that I think is clean.

In my point of view, your last commit 856609692ae2e8dadbb0b3d16f489a74a4f4c764 is only bloating the code without any benefit. Please revert this.

> I also think that else if should be if to cover the use case when someone wants to be super safe and run both checks.

If you like. Maybe that's the expected behaviour when all params are set. However, you should update the comments to match this behaviour. I actually adapted the behaviour to the behaviour originally promised in the comments...